docs/polaris/data-science-workflows/frameworks/rapids.md

```diff
@@ -0,0 +1,275 @@
+    export DASK_DISTRIBUTED__WORKER__MEMORY__Terminate="False"
+    
+    
+    # Setup scheduler
+    if [ "$ROLE" = "SCHEDULER" ]; then
+    
+      if [ "$CLUSTER_MODE" = "NVLINK" ]; then
+         CUDA_VISIBLE_DEVICES='0' DASK_UCX__CUDA_COPY=True DASK_UCX__TCP=True DASK_UCX__NVLINK=True DASK_UCX__INFINIBAND=False DASK_UCX__RDMACM=False nohup dask-scheduler --dashboard-address $DASHBOARD_PORT --protocol ucx --scheduler-file $SCHEDULER_FILE > $LOGDIR/$HOSTNAME-scheduler.log 2>&1 &
+      fi
+    
+      if [ "$CLUSTER_MODE" = "IB" ]; then
+         DASK_RMM__POOL_SIZE=1GB CUDA_VISIBLE_DEVICES='0' DASK_UCX__CUDA_COPY=True DASK_UCX__TCP=True DASK_UCX__NVLINK=True DASK_UCX__INFINIBAND=True DASK_UCX__RDMACM=False UCX_NET_DEVICES=mlx5_0:1 nohup dask-scheduler --dashboard-address $DASHBOARD_PORT --protocol ucx --interface ib0 --scheduler-file $SCHEDULER_FILE > $LOGDIR/$HOSTNAME-scheduler.log 2>&1 &
+      fi
+    
+      if [ "$CLUSTER_MODE" = "TCP" ]; then
+         CUDA_VISIBLE_DEVICES='0' nohup dask-scheduler --dashboard-address $DASHBOARD_PORT --protocol tcp --scheduler-file $SCHEDULER_FILE > $LOGDIR/$HOSTNAME-scheduler.log 2>&1 &
+      fi
+    fi
+    
+    
+    # Setup workers
+    if [ "$CLUSTER_MODE" = "NVLINK" ]; then
+        dask-cuda-worker --device-memory-limit $DEVICE_MEMORY_LIMIT --local-directory $LOCAL_DIRECTORY --rmm-pool-size=$POOL_SIZE --memory-limit=$MAX_SYSTEM_MEMORY --enable-tcp-over-ucx --enable-nvlink  --disable-infiniband --scheduler-file $SCHEDULER_FILE >> $LOGDIR/$HOSTNAME-worker.log 2>&1
+    fi
+    
+    if [ "$CLUSTER_MODE" = "IB" ]; then
+        dask-cuda-worker --device-memory-limit $DEVICE_MEMORY_LIMIT --local-directory $LOCAL_DIRECTORY --rmm-pool-size=$POOL_SIZE --memory-limit=$MAX_SYSTEM_MEMORY --enable-tcp-over-ucx --enable-nvlink  --enable-infiniband --disable-rdmacm --scheduler-file $SCHEDULER_FILE >> $LOGDIR/$HOSTNAME-worker.log 2>&1
+    fi
+    
+    if [ "$CLUSTER_MODE" = "TCP" ]; then
+        dask-cuda-worker --device-memory-limit $DEVICE_MEMORY_LIMIT --local-directory $LOCAL_DIRECTORY --rmm-pool-size=$POOL_SIZE --memory-limit=$MAX_SYSTEM_MEMORY --scheduler-file $SCHEDULER_FILE >> $LOGDIR/$HOSTNAME-worker.log 2>&1
+    fi
+    ```
+
+
+1. [Submit an interactive job](https://docs.alcf.anl.gov/polaris/running-jobs/) on `n` (here 2) nodes
+
+    ```bash
+    qsub -l select=2:system=polaris -l filesystems=home:eagle:grand -l walltime=00:30:00 -I -A YourProject -q YourQueue
+    ```
+
+    A shell opens up on one of the compute nodes
+
+
+1. Run the following script, `~/start_rapids_cluster_polaris.sh`. This will start the scheduler on one node and one worker per GPU on all nodes.  
+    
+    ```bash
+    #!/bin/bash
+
+    # start_rapids_cluster_polaris.sh
+    
+    NUM_NODES=$(cat $PBS_NODEFILE | wc -l)
+    TMP_EXE=tmp_rpds.sh
+    
+    cat > ${TMP_EXE} << EOF
+    #!/bin/bash
+    if [ \$PMI_RANK == 0 ]; then
+        ~/activate_rapids_env_polaris.sh ~/start_rapids_cluster_rank.sh SCHEDULER
```

Comment:
A trivial suggestion, for the starter scripts/directories (which are not created by the script but by the user) it'd be useful to have $PWD instead of tilde (~). I suspect most users would create these scripts inside a sub-directory and run it from there rather than directly inside the $HOME directory. An alternative is to specify a user-defined $INSTALLATION_DIR on top and use it everywhere.

Reply:
thoughts @FilippoSimini?